### ISSUE TITLE
Own change output from funding tx is not shown in the balance

### ISSUE THREAD

**Issue by holzeis** [get10101/10101#2286]
Labeling this as bug as it used to behave differently before.

**Comment by luckysori** [comment]
@holzeis: Do you mean that the change output is missing when the transaction is still unconfirmed? I can reproduce that, but once the transaction is confirmed, the balance is updated correctly.

We should still include pending outputs in the wallet balance somehow.

**Comment by holzeis** [comment]
> @holzeis: Do you mean that the change output is missing when the transaction is still unconfirmed?

Yes, we used to trust unconfirmed change output from our own transactions.

**Comment by luckysori** [comment]
I think this can be fixed by adding a `get_change_address` method to `rust-dlc`'s `Wallet` trait.

At the moment, there is only a `get_new_address` method which is used for every situation. `bdk` can trust unconfirmed payments to change addresses, but not unconfirmed payments to external addresses.
